backend/local/testing.go

```diff
@@ -50,39 +51,43 @@ func TestLocal(t *testing.T) (*Local, func()) {
 
 // TestLocalProvider modifies the ContextOpts of the *Local parameter to
 // have a provider with the given name.
-func TestLocalProvider(t *testing.T, b *Local, name string, schema *terraform.ProviderSchema) *terraform.MockResourceProvider {
-	t.Fatalf("TestLocalProvider is not yet updated to use the new provider types")
-	return nil
-	/*
-		// Build a mock resource provider for in-memory operations
-		p := new(terraform.MockResourceProvider)
-		p.GetSchemaReturn = schema
-		p.DiffReturn = &terraform.InstanceDiff{}
-		p.RefreshFn = func(
-			info *terraform.InstanceInfo,
-			s *terraform.InstanceState) (*terraform.InstanceState, error) {
-			return s, nil
-		}
-		p.ResourcesReturn = []terraform.ResourceType{
-			terraform.ResourceType{
-				Name: "test_instance",
-			},
+func TestLocalProvider(t *testing.T, b *Local, name string, schema *terraform.ProviderSchema) *terraform.MockProvider {
+	// Build a mock resource provider for in-memory operations
+	p := new(terraform.MockProvider)
+	p.GetSchemaReturn = schema
+	p.PlanResourceChangeFn = func(req providers.PlanResourceChangeRequest) providers.PlanResourceChangeResponse {
+		return providers.PlanResourceChangeResponse{
+			PlannedState:   req.ProposedNewState,
+			PlannedPrivate: req.PriorPrivate,
 		}
+	}
+	// p.DiffReturn = &terraform.InstanceDiff{}
```

Comment:
I haven't quite figured out if these need to be refactored back in

Reply:
I expect some of the tests are relying on these defaults while other tests are overriding them, but stubbing them out for now just to make it build is fine and we can figure out what's actually needed here when we review the individual tests.